### ISSUE TITLE
Google Fit upload issues

### ISSUE THREAD

**Issue by darander** [jonasoreland/runnerup#263]
Hi,

I tried to play around with Google Fit, just to see how the data would look like... However, I can't get it to upload anything. I've successfully added the Google Fit account, selected 1 set of training data, and press upload. It looks like it would succeeded, but nothing gets uploaded.

The short logcat is as follows:
I/org.runnerup(18733): Start uploading 1
V/WindowManager( 1761): not Base app: Adding window Window{3bd9d37f u0 Uploading to GoogleFit} at 16 of 23
I/org.runnerup.export.format.GoogleFitData(18733): Creating new dataSource: derived:com.google.location.bounding_box:973756715091:runnerup_location_summary
I/GoogleFit(18733): {"error":{"errors":[{"domain":"global","reason":"invalidArgument","message":"Bad Request"}],"code":400,"message":"Bad Request"}}
W/InputMethodManagerService( 1761): Window already focused, ignoring focus gain of: com.android.internal.view.IInputMethodClient$Stub$Proxy@2d69ef4c attribute=null, token = android.os.BinderProxy@3c328f12

Any ideas?

Oh, just forgot:
Nexus 4, running Android 5.1.1.
RunnerUp 1.44.

Cheers,
Anders

**Comment by paradix** [comment]
Does the activity shows up as synchronized with GoogleFit in runnerup?

**Comment by darander** [comment]
Hm, good question...

If I go into the history tab, they might actually show add uploaded. At
least they're greyed out as the other uploaders. (though they spell it out
add upload instead of uploaded).

If I go into configure account, they show up add of they could be uploaded.

Thus they might actually be uploaded, though I can't see them in Google
Fit, and the error code that I could see in the logcat.

**Comment by darander** [comment]
I just tried to re-upload the latest run (from yesterday), and I'm still not seeing anything i Google Fit. (I made this run with runtastic running in parallel on the phone, just to check how it works. The runtastic data is uploaded OK, and visible from fit.google.com).

A logcat does only reveal the same thing as earlier:

```
I/org.runnerup(18459): Start uploading 1
V/WindowManager( 1867): not Base app: Adding window Window{2889ba5c u0 Uploading to GoogleFit} at 16 of 23
I/org.runnerup.export.format.GoogleFitData(18459): Creating new dataSource: derived:com.google.location.bounding_box:973756715091:runnerup_location_summary
I/GoogleFit(18459): {"error":{"errors":[{"domain":"global","reason":"invalidArgument","message":"Bad Request"}],"code":400,"message":"Bad Request"}}
```

**Comment by paradix** [comment]
I'm still looking into this. It's a low progress :) but I can reproduce it with some activities of mine.

**Comment by darander** [comment]
Good to know that you can reproduce it, that's the most important part of fixing a bug... Then I'll keep my mouth shut (unless I can find some rare time and look into it myself).

Thanks for you effort!

**Comment by paradix** [comment]
@jonasoreland: could you please send me per email the current "developer project number" for RunnerUp

**Comment by jonasoreland** [comment]
what is your email again ??
can you email me your email address ??

/Jonas

**Comment by Almtesh** [comment]
Nothing new about this problem? I noticed it just now.

**Comment by jonasoreland** [comment]
nothing new...that I know of.

@paradix ??

/Jonas

**Comment by paradix** [comment]
To be honest, since I had some  running injuries I stopped looking into any sports tracker apps and also didn't fix this issue. I can't promise anything but I may look into it again at some point.

**Comment by Almtesh** [comment]
Ok, I put the runs manually, for the moment. I keep subscribed to this issue to get news.
If you want to test or get some debug, I can help you.

**Comment by yzerman19** [comment]
I just realized today that my 110+ runs in my history are not actually being uploaded to Google Fit, even though the RunnerUp app makes it seem like they are being uploaded.
I go to Google Fit on my computer and it shows I have 0 runs in my recent history.

Obviously I don't use Google Fit much, since this is the first time I noticed the uploads are not working, even though I have been "uploading" for about a year! Really the only reason I want to upload to Google Fit was so I had an online backup of my running history. I am thinking of wiping my phone, hence the reason I finally checked Google Fit to see if all my data is there (it's not!)

If the Google Fit upload cannot be fixed easily/soon, does anyone recommend a service I should upload to? There are many to choose from in the RunnerUp "Accounts" page..

FYI, My phone is: HTC One M8 (AT&T), OS: Android 5.0.2

Thanks!

**Comment by Marx2** [comment]
Google fit is the worst choice to use it as a backup. It's unreliable and it's interface is horrible. I didn't see any export function.
Use Strava as a backup. You can easily export all your activities as standard tracks from their website. 
Additionally you can install Stavistix plugin in chrome which gives you many additional statistics.

**Comment by yzerman19** [comment]
If Google Fit upload doesn't work, I think it would be best to display an error message when the upload fails. Or maybe remove it entirely. I have been "uploading" for a year thinking that it was working. There is no error message.

Now that I am uploading to Google Fit plus one other service, I see on the Upload tab of each of my activities that Google Fit says "upload" while the other service says "Uploaded". This is very, very subtle way of telling me that Google Fit didn't upload and the other service did upload already. Maybe that status should say "upload failed" for Google Fit?

**Comment by Marx2** [comment]
Google fit has serious bugs, for example no priorities, so last activity at the same time overrides all previous. It causes many problems with mi band for example. It doesn't mean that it doesn't work. It simply means it's buggy and you can't trust it, and it cannot be used as a backup.
But having connector is good because in future it probably will start to work properly.

**Comment by oylenshpeegul** [comment]
I tried RunnerUp this morning for the first time. Seemed to work great! I configured it to upload to Google Fit and it acted like it worked, but logging in to Fit showed no activities. I thought maybe it took time to sync, so I left it for the day. Tonight there were still no activities, so I configured it to upload to Runalyze and it also acted like it worked. However, logging in to Runalyze showed my run from this morning. That's when I did a little searching and found this Issue.

So put me down for a "me too" on this issue. Everything about runnerup is working great except for upload to Google Fit. I've never used Google Fit or Runalyze, so I don't really care; I'm happy to use Runalyze instead.
